Correcting a session problem.

Bakrog opened the pull request:
Correcting a problem when sometimes you can get a new session in your controller.

Commit:
Correcting a session problem.

Turini commented:
@Bakrog, thanks for doing it :)
@linyatis when/if possible, can you test the solution?

linyatis commented:
@Bakrog and @Turini, I'll try this tonight!

linyatis commented:
@Turini, It does not work with annotation, just programmatically =(
@Bakrog, you did exactly the same thing as I did.  
The session and subject now work together, but not at Shiro's context. Somehow is not the same instance that Shiro uses.

Bakrog commented:
@linyatis, Oh, I didn't try using annotations. I'll look at this tonight.

Bakrog commented:
@Turini, Can I add a provided dependency to servlet-api 3.0.1?

Turini commented:
hum, did you mean `javax-servlet-api`? VRaptor's core uses the 
3.1.0 version, we really need the 3.0.1? (it will help somehow here?)

Bakrog committed:
Correcting shiro security annotations problem and this closes
dipold/vraptor-shiro#4.
Now the "interceptors" tag in beans.xml need to be declared in your
dynamic web project.

Bakrog commented:
@Turini never mind! I changed to javax.ejb-api and I could correct issue #4.
@linyatis, @mtnoronha, when/if possible, can you test the solution?
P.S: You can't use SecurityUtils.getSubject(). I'm trying to figure out how build the subject and associate it with the current thread.

Bakrog committed:
Removing a sysout.

linyatis commented:
@Bakrog, I can't test it now, but let me understand. You fixed issue #4 by now? Do you fixed the issue of the session using it programmatically and with annotations? What do you need that I test?

By the way, @Turini, we never opened an issue about this (session), right?
Would be interesting open it or not?

Bakrog commented:
@linyatis, yes, I correct issue #4 by adding the annotation Startup (javax.ejb-api) in the class SecurityFacade.
Now you can use apache shiro annotations or do it programmatically. But you need to inject the subject and cant use SecurityUtils.getSubject() because there is another thread for shiro and the subject isnt build in that thread. And you need to declare the interceptor in your dynamic web project.
Can you test if the annotation problem is solved in your project?

Bakrog commented:
Oh, you only need to inject the subject if you're doing it programmatically.

Turini commented:
> we never opened an issue about this (session), right?

Not sure, but I think we ddidn't. 

> Would be interesting open it or not?

Yes, it would be nice.

linyatis commented:
Hey guys! Sorry for the delay in responding, I had forgotten!
@Bakrog, I tested here and looks like issue #4 is corrected now.

@Turini, later I will open that issue about annotations not working with Wildfly. I will test with others containers.

Bakrog commented:
@linyatis @Turini I think issue #5 is about the annotations problem.
@linyatis Did the annotations not work with this pull request? Did you declare the interceptor (br.com.caelum.vraptor.security.interceptor.SecurityInterceptor) in your project?
